commit 119294278af2390971305224c9772d89525d97e1
Author: Justus Winter <4winter@informatik.uni-hamburg.de>
Date:   Sat Apr 5 15:08:06 2014 +0200

    include: make the notify_port_t types translation functions mutable
    
    Make the intran, outtran and destructor functions mutable using
    preprocessor macros.  Make it possible to inject imports using the
    NOTIFY_IMPORTS macro.  This way, userspace servers can provide their
    own translation functions.
    
    * include/mach/notify.defs: Honor NOTIFY_IMPORTS.
    (notify_port_t): Make the translation mutable using preprocessor macros.

diff --git a/include/mach/notify.defs b/include/mach/notify.defs
index e06f6b4..fdf35e9 100644
--- a/include/mach/notify.defs
+++ b/include/mach/notify.defs
@@ -28,6 +28,10 @@ subsystem notify 64;
 
 #include <mach/std_types.defs>
 
+#ifdef NOTIFY_IMPORTS
+NOTIFY_IMPORTS
+#endif
+
 #if	SEQNOS
 serverprefix do_seqnos_;
 serverdemux seqnos_notify_server;
@@ -37,7 +41,17 @@ serverdemux notify_server;
 #endif	SEQNOS
 
 type notify_port_t = MACH_MSG_TYPE_MOVE_SEND_ONCE
-	ctype: mach_port_t;
+	ctype: mach_port_t
+#ifdef	NOTIFY_INTRAN
+	intran: NOTIFY_INTRAN
+#endif
+#ifdef	NOTIFY_OUTTRAN
+	outtran: NOTIFY_OUTTRAN
+#endif
+#ifdef	NOTIFY_DESTRUCTOR
+	destructor: NOTIFY_DESTRUCTOR
+#endif
+;
 
 /* MACH_NOTIFY_FIRST: 0100 */
 skip;
